src/llvm/di.rs

```diff
@@ -404,6 +438,58 @@ impl DISanitizer {
 
         replace
     }
+
+    fn di_compile_units(&self) -> Vec<DICompileUnit> {
+        let compile_unit_name = c"llvm.dbg.cu";
+
+        // Get the number of DICompileUnit.
+        let num_di_cu =
+            unsafe { LLVMGetNamedMetadataNumOperands(self.module, compile_unit_name.as_ptr()) };
+
+        // Create a vector to hold them all.
+        let mut di_cus: Vec<LLVMValueRef> = vec![core::ptr::null_mut(); num_di_cu as usize];
+
+        unsafe {
+            LLVMGetNamedMetadataOperands(
+                self.module,
+                compile_unit_name.as_ptr(),
+                di_cus.as_mut_ptr(),
+            )
+        };
+
+        di_cus
+            .into_iter()
+            .map(|v| unsafe { DICompileUnit::from_value_ref(v) })
+            .collect()
+    }
+
+    // Removes `c_void` Rust enum from all DICompileUnit.
+    // After replacing `c_void` enum with a DIBasicType
+    // the DICompileUnit still references the `c_void` enum which
+    // triggers a casting assertion in LLVM.
```

Comment:
Please make it a proper docstring and add links to types:

```suggestion
    /// Removes `c_void` Rust enum from all [`DICompileUnit`].
    /// After replacing `c_void` enum with a [`DIBasicType`]
    /// the [`DICompileUnit`] still references the `c_void` enum which
    /// triggers a casting assertion in LLVM.
```

Also, the lines are a bit short, ~60 characters. Feel free to use up to ~80 per line.

Reply:
Done